### ISSUE TITLE
Somehow ended up with partial node_modules in instaclone cache

### ISSUE THREAD

**Issue by marcello3d** [vivlabs/instaclone#16]
We hit a case where instaclone ended up with a bad repo. Not sure how it happened, appears to be a partial tar unzip—maybe it didn't download properly/failed to unzip/was interrupted.

Maybe we should untar to a temporary folder and move once it's done.

Worokaround: `instaclone purge` and try again.

**Comment by mblakele** [comment]
One thing we didn't check for was a full disk. That would interfere with the untar operation.

**Comment by jlevy** [comment]
In general the intention was to have all operations write to a temp folder, then get renamed on success, to avoid stuff like this. This would mean full disk would leave a temp dir, but it would have a random name that doesn't conflict. See https://github.com/vivlabs/instaclone/blob/master/instaclone/instaclone.py#L105
Did that use of atomic_output_file not work somehow?

**Comment by mblakele** [comment]
@jlevy we don't have any details of the error, only the aftermath: files missing from node_modules. The workaround `instaclone purge` destroyed any forensic data.

**Comment by jlevy** [comment]
Yeah. :disappointed: If desired, you probably could reproduce by putting the .instaclone dir on a very small disk image and rerunning.

**Comment by mblakele** [comment]
`dd if=/dev/zero of=/tmp/test bs=1048576 count=123456 conv=sparse` is pretty fast. However I think we can rule out disk space because OSX presents prominent warnings about that. Anyway here's what happens:

```
$ instaclone install --copy
using config file: ./instaclone.yml
cache dir not found, so creating: /Users/mblakele/.instaclone/cache
error: [Errno 28] No space left on device: '/Users/mblakele/.instaclone/cache'
(run with --debug for traceback info)
-bash: test: -eq: unary operator expected
```
